File: 3208-alternating-groups-ii/3208-alternating-groups-ii.py
class Solution:
    def numberOfAlternatingGroups(self, colors: List[int], k: int) -> int:
        if len(colors) < k:
            return 0
        colors = colors + colors[:k - 1]
        n = len(colors)
        l = 0
        curr = 0
        result = 0
        while curr < n - 1: 
            x = curr + 1
            if colors[x] ^ colors[curr]:
                if x - l > k - 1:
                    l += 1
            else:
                l = x
            curr = x
            if curr - l == k - 1:
                result += 1
        return result

File: 3208-alternating-groups-ii/3208-alternating-groups-ii.go
func numberOfAlternatingGroups(colors []int, k int) int {
    colors = append(colors, colors[:k - 1]...)
    var l, curr, result int
    var n int = len(colors)
    for curr < n - 1 {
        x := curr + 1
        if colors[x] ^ colors[curr] == 1 {
            if x - l > k - 1 {
                l ++
            } 
        } else {
            l = x
        }
        curr = x
        if curr - l == k - 1 {
            result ++
        }
    }
    return result
}